File: Codes/2178-maximum-split-of-positive-even-integers.py
'''
Author: LetMeFly
Date: 2023-07-06 07:45:05
LastEditors: LetMeFly
LastEditTime: 2023-07-06 07:46:14
'''
from typing import List

class Solution:
    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        if finalSum % 2:
            return []
        ans = []
        now = 2
        while finalSum - now > now:
            ans.append(now)
            finalSum -= now
            now += 2
        ans.append(finalSum)
        return ans

File: Codes/2178-maximum-split-of-positive-even-integers.cpp
/*
 * @Author: LetMeFly
 * @Date: 2023-07-06 07:41:32
 * @LastEditors: LetMeFly
 * @LastEditTime: 2023-07-06 07:44:35
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

typedef long long ll;

class Solution {
public:
    vector<ll> maximumEvenSplit(ll finalSum) {
        if (finalSum % 2) {
            return {};
        }
        vector<ll> ans;
        ll now = 2;
        while (finalSum - now > now) {
            ans.push_back(now);
            finalSum -= now;
            now += 2;
        }
        ans.push_back(finalSum);
        return ans;
    }
};